consensus/module.go

```diff
@@ -58,8 +61,16 @@ type consensusModule struct {
 	MaxBlockBytes uint64
 }
 
-func Create(cfg *genesis.Config, genesis *genesis.GenesisState) (modules.ConsensusModule, error) {
-	leaderElectionMod, err := leader_election.Create(cfg, genesis)
+func Create(config, gen json.RawMessage) (modules.ConsensusModule, error) {
```

Comment:
Not a fan of this design.

I think running `InitConfig` and `InitiGensis` is something we are going to end up doing in every module, and given that it's an app specific blockchain, I don't see why we shouldn't just use the specific `Config` or `GenesisState` objects.

Reply:
Shared can't import the objects without cyclical dep, so we'll need a specific integration package to do so.

This design is from the Cosmos sdk and it works well for V0

Reply:
Discussed offline and outdated, resolving.